src/transformStateWithClones.js

```diff
@@ -8,6 +8,23 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+    for (const key in actions) {
+    switch (actions[key].type) {
+      case 'addProperties':
+        for (const extraKey in actions[key].extraData) {
+          state[extraKey] = actions[key].extraData[extraKey];
+        }
+        break;
+      case 'removeProperties':
+        for (const extraKey in actions[key].keysToRemove) {
+          delete state[actions[key].keysToRemove[extraKey]];
```

Comment:
Similar to the 'addProperties' case, ensure that you are working with a clone of the state object when removing properties.